### DISCUSSION TITLE
Proposals based on usage experience at silent.link

### DISCUSSION THREAD

**Discussion by silent-link** [btcpayserver/btcpayserver#5330] [Ideas / Feature Requests]
1. Payouts in LN (currently not working)

`https://docs.btcpayserver.org/api/v1/pull-payments/{pullPaymentId}/payouts`

does not accept "LN" as `payment-method`.
When using "Off-chain" as  `payment-method` with LN invoice for `destination, no payout is created.

2. Payouts in XMR (by calling an external script)

We understand that BTCPayServer XMR wallet is watch-only.
However, it would be trivial to run a CLI XMR wallet from a script.
We suggest that BTCPayServer should be able to run such script with relevant parameters.

It is absolutely crucial to be able to make payouts in the same asset as was used for the incoming payment, as BTCPayServer would become an exchange service from one asset into another (XMR to BTC), that is not desirable for any merchant.

3. Same for creation of pull payments.

4. Different UTXO collection strategies when making outbound on-chain payments:

- collect smallest UTXOs (consolidation)
- collect UTXO in the range of amounts
- do not create change by sacrificing to miners but still sending the precise amount
- **"Select All" control**. We have to select dozens of UTXO one by one. Select All might be applied to all filtered UTXO or all UTXO visible on the page.
- 
  We usually want to select only invoice UTXO and avoid selecting change and manually received UTXO, those should be processed manually.

  It would make sense to target a certain maximum number of UTXO for the given amount, as Specter-DIY cannot sign more than a ~100 UTXO in a single PSBT.

5. Fault tolerant, redundant deployment as a default.

As far as we understand any merchant would prefer to pay for 1-3 extra VPS to have a fault tolerant system. 
That is especially important for lightning nodes that are quite hard to recover in case of loss of a VPS.
It is very desirable to be able to run backup VPS on different hosting providers,
As far as we know, there are existing solutions to "clusterize" LN, at least for CLN.

Att. @Kukks

**Comment by dennisreimann** [comment]
Regarding point 1: For the [Create Payout](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/PullPayments_CreatePayout) call you need to specify `paymentMethod` as `BTC_LightningNetwork`. We'll add a list of those to the docs, I just saw that we haven't documented it properly.

**Reply by silent-link** [reply]
Could you please clarify, what should we specify when creating a Pull Payment for:
- LN
- XMR
as per https://docs.btcpayserver.org/API/Greenfield/v1/#operation/PullPayments_CreatePullPayment

**Reply by dennisreimann** [reply]
You'd need to pass the `paymentMethod` parameter with the value `BTC_LightningNetwork` for Lightning and `XMR` for Monero.

**Reply by silent-link** [reply]
any chance we can use monero CLI wallet to make outbound payments?

**Reply by dennisreimann** [reply]
Right now the Monero wallet is watch-only. You might want to approach people from the community that could add the functionality, maybe @sethforprivacy can help.

The core team does not develop and maintain the altcoin modules, this is up to the community of the particular altcoin.

**Reply by silent-link** [reply]
any chance BTCPayServer could start a shell script in this case, passing the payout address and amount?

**Reply by Kukks** [reply]
It would be part of a different feature: Webhooks for payouts

**Reply by silent-link** [reply]
eta? it is vital for our next service that you all will like

**Reply by Kukks** [reply]
I can't provide an ETA but webhooks for non invoice features are high on our list due to popular requests.
